presto-spi/src/main/java/com/facebook/presto/spi/block/ByteArrayBlock.java

```diff
@@ -158,6 +158,11 @@ public Block copyRegion(int positionOffset, int length)
     {
         checkValidRegion(getPositionCount(), positionOffset, length);
 
+        if (positionOffset == 0 && length == positionOffset &&
+                arrayOffset == 0 && positionCount == values.length) {
+            return this;
```

Comment:
In the comment to `copyRegion` it says

```
     The region returned must be a compact representation of the original block, 
unless their internal representation will be exactly the same.
```

But I agree now `copyRegion` might be slightly vague.  I note both `getRegion` and `copyRegion` will return a slice of the block, but `copyRegion` guaranteed the returned block is compact. So one potential name is `getRegionAsCompactBlock`? 

Alternatively, given almost all call to `copyRegion` copy the whole block (except 1 place in code IIRC), we can have `getRegion` and `compact`. So the current funcitonlaity of `block.copyRegion()` becomes `block.getRegion().compact()`.